sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/SupportsPushDownTopN.java

```diff
@@ -31,8 +31,23 @@
 @Evolving
 public interface SupportsPushDownTopN extends ScanBuilder {
 
+    /**
+     * The first element of tuple means topN can be pushed down to data source.
+     * The second element of tuple is false means Spark complete push down topN,
+     * otherwise means Spark partial push down topN and need preserves sort too.
+     */
+    class Tuple {
+        public final Boolean enable;
+        public final Boolean isPartial;
```

Comment:
2 boolean means we should have two methods... You did it right at the beginning, why do you change the API?